be/test/olap/rowset/segment_v2/segment_test.cpp

```diff
@@ -0,0 +1,259 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "olap/rowset/segment_v2/segment.h"
+#include "olap/rowset/segment_v2/segment_writer.h"
+#include "olap/rowset/segment_v2/segment_iterator.h"
+
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include "common/logging.h"
+#include "olap/olap_common.h"
+#include "olap/row_cursor.h"
+#include "olap/tablet_schema.h"
+#include "olap/row_block.h"
+#include "olap/row_block2.h"
+#include "olap/types.h"
+#include "olap/tablet_schema_helper.h"
+#include "util/file_utils.h"
+
+namespace doris {
+namespace segment_v2 {
+
+class SegmentReaderWriterTest : public testing::Test {
+public:
+    SegmentReaderWriterTest() { }
+    virtual ~SegmentReaderWriterTest() {
+    }
+};
+
+TEST_F(SegmentReaderWriterTest, normal) {
+
+    size_t num_rows_per_block = 10;
+
+    std::shared_ptr<TabletSchema> tablet_schema(new TabletSchema());
+    tablet_schema->_num_columns = 4;
+    tablet_schema->_num_key_columns = 3;
+    tablet_schema->_num_short_key_columns = 2;
```

Comment:
What's the difference between `_num_key_columns` and `_num_short_key_columns`, when could them be unequal?

Reply:
We use some of keys as index, which is `short` key. We don't use all key columns as key for memory concern. We want to load all index in to memory to accelerate reading.

Reply:
Now I understand, thanks